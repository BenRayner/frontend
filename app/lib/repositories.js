// @flow

export function repositoryProviderIcon(providerName: ?string): string {
  switch (providerName) {
    case 'RepositoryProviderBitbucket':
      return 'bitbucket';
    case 'RepositoryProviderGithub':
      return 'github';
    case 'RepositoryProviderGithubEnterprise':
      return 'github';
    case 'RepositoryProviderGitlab':
      return 'gitlab';
    case 'RepositoryProviderGitlabCommunity':
      return 'gitlab';
    default:
      return 'git';
  }
}
