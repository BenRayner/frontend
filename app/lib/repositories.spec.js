/* global describe, it, expect */
import { repositoryProviderIcon } from './repositories';

const PROVIDERS = [
  'RepositoryProviderBitbucket',
  'RepositoryProviderCodebase',
  'RepositoryProviderGithub',
  'RepositoryProviderGithubEnterprise',
  'RepositoryProviderGitlab',
  'RepositoryProviderGitlabCommunity',
  'RepositoryProviderUnknown'
];

describe('repositoryProviderIcon', () => {
  describe('returns expected icon names', () => {
    PROVIDERS.forEach((provider) => {
      it(`for ${provider}`, () => {
        expect(repositoryProviderIcon(provider)).toMatchSnapshot();
      });
    });
  });
});
