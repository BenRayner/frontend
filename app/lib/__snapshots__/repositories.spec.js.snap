// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderBitbucket 1`] = `"bitbucket"`;

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderCodebase 1`] = `"git"`;

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderGithub 1`] = `"github"`;

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderGithubEnterprise 1`] = `"github"`;

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderGitlab 1`] = `"gitlab"`;

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderGitlabCommunity 1`] = `"gitlab"`;

exports[`repositoryProviderIcon returns expected icon names for RepositoryProviderUnknown 1`] = `"git"`;
